app/pages/lab/social-links-editor.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import DragReorderable from 'drag-reorderable';
+import AutoSave from '../../components/auto-save.coffee';
+import SOCIAL_ICONS from '../../lib/social-icons';
+
+export default class SocialLinksEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    const socialOrder = Object.keys(SOCIAL_ICONS).map(key => key);
+    this.state = {
+      socialOrder
+    };
+    this.reorderDefault = this.reorderDefault.bind(this);
+    this.handleNewLink = this.handleNewLink.bind(this);
+    this.handleLinkReorder = this.handleLinkReorder.bind(this);
+    this.handleRemoveLink = this.handleRemoveLink.bind(this);
+    this.renderRow = this.renderRow.bind(this);
+  }
+
+  componentDidMount() {
+    this.reorderDefault();
+  }
+
+  reorderDefault() {
+    const socialUrls = this.props.project.urls.filter(url => url.path);
+    const newOrder = [];
+    socialUrls.map(link => newOrder.push(link.site));
+    this.state.socialOrder.map((item) => {
+      if (newOrder.indexOf(item) < 0) {
+        newOrder.push(item);
+      }
+    });
+    this.setState({ socialOrder: newOrder });
+  }
+
+  handleNewLink(site, e) {
+    let index = this.indexFinder(this.props.project.urls, site);
+    if (index < 0) { index = this.props.project.urls.length; }
+
+    if (e.target.value) {
+      const changes = {
+        [`urls.${index}`]: {
+          label: '',
+          path: e.target.value,
+          site,
+          url: `https://${site}${e.target.value}`
+        }
+      };
+      this.props.project.update(changes);
+    } else {
+      this.handleRemoveLink(site);
+    }
+  }
+
+  handleLinkReorder(newLinkOrder) {
```

Comment:
That's a great solution. Thanks a ton for the suggestion. This was one part of the code that I wracked my brain about, so it's good that it can be refactored into the few lines above.